Runtime/Scripts/RGBotConfigs/RGButtonState.cs

```diff
@@ -1,26 +1,35 @@
 using System.Collections.Generic;
+using RegressionGames.StateActionTypes;
 using UnityEngine;
 using UnityEngine.UI;
 
 namespace RegressionGames.RGBotConfigs
 {
-    public class RGButtonState : RGState
+    public class RGButtonState : MonoBehaviour, IRGState
     {
+        [Tooltip("A type name for associating like objects in the state")]
+        public string objectType = "Button";
 
-        public RGButtonState()
-        {
-            if (string.IsNullOrEmpty(rgEntity.objectType))
-            {
-                rgEntity.objectType = "Button";
-            }
-        }
-
-        public override Dictionary<string, object> GetState()
+        public Dictionary<string, object> GetState()
```

Comment:
this method is left over from when it extended RGState.  I think for buttons we can just move this code into `GetGameObjectState`.

Reply:
Fixed